### ISSUE TITLE
bug:  casting inferred type to hint module will generate error

### ISSUE THREAD

**Issue by SolaWing** [castwide/solargraph#424]
```ruby
module Interface
end

class A
  include Interface
end

# @type [Interface]
a = A.new # <-- [Typecheck] [E] Declared type Interface does not match inferred type A for variable a
```

solargraph.yml
```yml
---
include:
- "**/*.rb"
exclude:
- spec/**/*
- test/**/*
- vendor/**/*
- ".bundle/**/*"
require: []
domains: []
reporters:
- require_not_found
- typecheck:typed
require_paths: []
plugins: []
max_files: 5000
```

**Comment by castwide** [comment]
Confirmed. The fix is in the master branch and will be included in the next release.

**Comment by castwide** [comment]
Released in v0.41.0.
